pos_accounting/src/pool/tests/delta_tests.rs

```diff
@@ -162,8 +163,125 @@ fn check_merge_delta() {
     let undo_data = delta1.merge_with_delta(delta2.consume()).unwrap();
     assert_eq!(delta1.data(), &expected_data);
 
-    delta1.undo_delta_merge(delta2_origin_data, undo_data).unwrap();
+    delta1.undo_delta_merge(undo_data).unwrap();
     assert_eq!(delta1.data(), &delta1_origin_data);
 }
 
+#[test]
+fn check_merge_values_with_deltas() {
+    let (_, pub_key1) = PrivateKey::new(KeyKind::RistrettoSchnorr);
+    let (_, pub_key2) = PrivateKey::new(KeyKind::RistrettoSchnorr);
+
+    let mut storage = InMemoryPoSAccounting {
+        pool_data: BTreeMap::from([(
+            H256::from_low_u64_be(1),
+            PoolData::new(pub_key1.clone(), Amount::from_atoms(100)),
+        )]),
+        pool_balances: BTreeMap::from([
+            (H256::from_low_u64_be(3), Amount::from_atoms(100)),
```

Comment:
Two things:
1. Maybe some variation in values
2. You're probably making everything public to initialize like this. How about initializing with some operations and ramping up the values, testing their undos incrementally?